tailscale/resource_tailnet_key.go

```diff
@@ -76,6 +82,7 @@ func resourceTailnetKeyCreate(ctx context.Context, d *schema.ResourceData, m int
 	ephemeral := d.Get("ephemeral").(bool)
 	preauthorized := d.Get("preauthorized").(bool)
 	expiry, hasExpiry := d.GetOk("expiry")
+	description := d.Get("description").(string)
```

Comment:
Perhaps use a GetOk here, with a separate hasDescription bool that can be used as part of the condition below (instead of len)? This will make it more consistent with how we handle `expiry`.